xapian-applications/omega/docs/omegascript.rst

```diff
@@ -418,6 +418,11 @@ $prettyurl{URL}
 	URL for displaying to the user (rather than in links), but which should
 	still work if copied and pasted.
 
+$qid{TEXT}
+	unique ID for the query TEXT which would be usually taken from $query.
+	This command uses MD5 hashing algorithm to produce a 128-bit hash for
+	the given query text which is used as a unique ID.
```

Comment:
I'm not keen to tie the implementation to using MD5 in the documentation like this does.  MD5 is not a secure hash today, and organisations may (with good intent, though arguably somewhat misguidedly) impose a policy that MD5 should not be used at all.  As this stands, it would be hard for us to move `$qid` to using a more modern hash, because the documentation explicitly says that it uses MD5.  If we're going to pick a particular hash like this, MD5 is not the right choice.

Also, the name `$qid` is rather cryptic and quite specific to this particular use - where possible we prefer to add more generic functions which can be used to achieve a specific requirement but have wider applicability, and the space trimming makes `$qid` less useful as a generic way to hash something.

So I think it would make more sense to add a new omegascript command which calculates the hash of a given text string, and to keep that command generic, any normalising you want to do is left to the code which makes use of that.  So instead of `$qid{$query}` you might write:

```
$md5{$transform{$query,^ *(.*?) *$,\1}}
```

Or perhaps `$hash{md5,$transform{$query,^ *(.*?) *$,\1}}` so it's easier to add new hash algorithms and configure them from the template.